components/fxa-client/src/ffi.rs

```diff
@@ -36,6 +39,18 @@ pub mod error_codes {
     pub const NETWORK: i32 = 3;
 }
 
+pub trait FromProtobufPtr<T: prost::Message + Default + Into<Self>>: Sized {
```

Comment:
Thought this can help, since I basically did the same thing in #3264

Reply:
+1; I also wonder if it's the sort of thing we should push all the way down into the `ffi-support` create, /cc @thomcc

Reply:
I think this could be a generic function instead of a trait, no?

Regardless, my hesitation with putting this into ffi-support is: I don't really like adding an unsafe API like that. In particular, this shape of API (which we admittedly use elsewhere) has bitten us in the past (and each time we've said something along the lines of "lets be less aggressive about it" but then did nothing...).

As it is, I don't believe FFI support offers any non-deprecated unsafe APIs that you're expected to use directly, and I think that's a good thing -- We've had very little issue with the APIs it offers in this way.

The other thing is that we don't have a prost dependency in ffi-support atm, but I guess it could be kept optional.

Anyway, it's also not like a safe api for this can't be done, it's just not really trivial. That said, I'm pretty surprised that we haven't gotten around to it by now... So, given those things, I'd rather not add it -- it'd be the first unsafe user-facing API we add to the crate, and really we should bite the bullet and just make a safe API for it, or wait for uniffi to save us.

Reply:
Good points, and I don't feel strongly about needing to move it with any urgency.

Is there a world where we could wrap it into some sort of safely-ecapsulated macro, like an `implement_from_ffi_by_protobuf` that is dual to `implement_into_ffi_by_protobuf`?